distribution/errors.go

```diff
@@ -213,6 +213,12 @@ func (e reservedNameError) Error() string {
 
 func (e reservedNameError) Forbidden() {}
 
+const deprecatedSchema1 = "[DEPRECATION NOTICE] Docker Image Format v1 and Docker Image manifest version 2, schema 1 support is disabled by default and will be removed in an upcoming release. More information at https://docs.docker.com/go/deprecated-image-specs/"
+
+type invalidArgumentErr struct{ error }
+
+func (invalidArgumentErr) InvalidParameter() {}
+
 func DeprecatedSchema1ImageMessage(ref reference.Named) string {
-	return fmt.Sprintf("[DEPRECATION NOTICE] Docker Image Format v1, and Docker Image manifest version 2, schema 1 support will be removed in an upcoming release. Suggest the author of %s to upgrade the image to the OCI Format, or Docker Image manifest v2, schema 2. More information at https://docs.docker.com/go/deprecated-image-specs/", ref)
+	return fmt.Sprintf("[DEPRECATION NOTICE] Docker Image Format v1 and Docker Image manifest version 2, schema 1 support is disabled by default and will be removed in an upcoming release. Suggest the author of %s to upgrade the image to the OCI Format or Docker Image manifest v2, schema 2. More information at https://docs.docker.com/go/deprecated-image-specs/", ref)
```

Comment:
nit:

Maybe we could rewrite this into:
```suggestion
    msg := "[DEPRECATION NOTICE] Docker Image Format v1 and Docker Image manifest version 2, schema 1 support is disabled by default and will be removed in an upcoming release."
    if ref != nil {
        msg += " Suggest the author of " + ref.String() + " to upgrade the image to the OCI Format or Docker Image manifest v2, schema 2."
    }
	return msg + " More information at https://docs.docker.com/go/deprecated-image-specs/"
```

and then just use `DeprecatedSchema1ImageMessage(nil)` instead of duplicating the message in `deprecatedSchema1`?

Reply:
Oh! 🤦 I initially wanted to do so, but thought I'd had to change the signature to accept a pointer, but that's not the case. Yes, I can change that for sure; let me do so.